### PR TITLE
Clean up sandbox-on-x conformance tests.

### PR DESCRIPTION
**Pull request by mziolekda**
Make sure that the unsettled state of the ledger does not spill over to the subsequent test. 
Without that stabilization the subsequent test may pick-up noise and fail for all the wrong reasons.

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ConfigManagementServiceIT.scala [hunk 1]
```diff
@@ -71,6 +71,10 @@ final class ConfigManagementServiceIT extends LedgerTestSuite {
           newTimeModel = oldTimeModel,
         )
         .mustFail("setting a time model with an expired MRT")
+
+      // Above operation finished on a timeout, but may still succeed asynchronously.
+      // Stabilize the ledger state before leaving the test case.
+      _ <- stabilize(ledger)
```
Comment (nmarton-da) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ConfigManagementServiceIT.scala:77]: but here we still have to read after stabilizing, and making sure that what we read is not the failed one (as that one will fail anyway as we discussed)
or is this what you wanted to have a separate PR for?

**Review by nmarton-da — COMMENTED** [review]
LGTM

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ConfigManagementServiceIT.scala (hunk 1 above)
Comment (rautenrieth-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ConfigManagementServiceIT.scala:77]: I also don't understand why we wait for the participant to catch up, but don't read from it afterwards.

Comment (miklos-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ConfigManagementServiceIT.scala:77]: @mziolekda Do you want to move this back to draft?

Comment (mziolekda) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ConfigManagementServiceIT.scala:77]: This is making sure that the unsettled state of the ledger does not spill over to the subsequent test. Without stabilizing the subsequent test may pick-up noise and fail for all the wrong reasons.

Comment (mziolekda) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ConfigManagementServiceIT.scala:77]: Oh I see where the confusion comes from, allocateParty is a write+read operation. It waits until party allocation arrives in teh index-db.

Comment (rautenrieth-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ConfigManagementServiceIT.scala:77]: > making sure that the unsettled state of the ledger does not spill over to the subsequent test

I see. `setTimeModel` is _also_ a read+write operation, blocking until the change was witnessed in the index db. However, the last call to `setTimeModel` is using an already expired MRT, which means the participant will immediately abort waiting for the confirmation of the change.

You also just swap the "Restore the original time model" and "Try to set a time model with an expired MRT." blocks. The "Restore the original time model" call will serve the same purpose as the `stabilize` call.

**Review by rautenrieth-da — APPROVED** [review]
The change makes sense, left a comment about solving the spill over state without adding the `stabilize` method.

**Pull request merged**